x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/popover_editor.tsx

```diff
@@ -56,6 +56,31 @@ function asOperationOptions(operationTypes: OperationType[], compatibleWithCurre
     }));
 }
 
+const LabelInput = ({ value, onChange }: { value: string; onChange: (value: string) => void }) => {
+  const [inputValue, setInputValue] = useState(value);
+
+  useEffect(() => {
+    setInputValue(value);
+  }, [value, setInputValue]);
+
+  const onChangeDebounced = useMemo(() => _.debounce(onChange, 256), [onChange]);
```

Comment:
We have a hook you could replace this with called `useDebounce`, via `import { useDebounce } from 'react-use';`. The main difference is that that hook clears itself on unmount.